npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -492,6 +494,18 @@ def handle_npm_updater_error(error)
             raise Dependabot::PrivateSourceTimedOut, T.must(msg)
           end
 
+          # Error handled when no authentication info ( _auth = user:pass )
+          # is provided in config file (.npmrc) to access private registry
+          if error_message.match?(UNABLE_TO_AUTH_NPMRC)
+            msg = "check .npmrc config file"
```

Comment:
please can we get unit tests covering both if branches (for both possible error cases)

Reply:
@abdulapopoola adds test cases for error types.